Packages/ClientRuntime/Sources/Waiters/Matcher.swift

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+extension Acceptor {
+
+    /// `Acceptor.Matcher` is a Swift-native equivalent of Smithy matchers:
+    /// https://awslabs.github.io/smithy/2.0/additional-specs/waiters.html#matcher-union
+    public enum Matcher {
```

Comment:
Nit: Similar to a comment above, it'd be helpful to provide a description as to what a _Matcher_ is or what is does